crates/bevy_utils/src/option_ext.rs

```diff
@@ -0,0 +1,90 @@
+//! Extensions to the [`Option`] type used in Bevy.
```

Comment:
I think these should live in `bevy_ecs::error`. Currently our "boxed error type" is "just" `core::error::Error`, but there is a _very_ good chance that we will be swapping that for something custom (or anyhow). The `assume` coercion will be tied directly to the bevy ecs error type, so it should live there.

Reply:
Resolved in [210faa7](https://github.com/bevyengine/bevy/pull/17739/commits/210faa7418348289342a517c183eca8e6f7c30c7).